src/isolineFeature.js

```diff
@@ -0,0 +1,891 @@
+    var val = value.value,
+        lowhigh = Array(mesh.value.length),
+        chains = [],
+        i, v, pattern, lines;
+    /* Determine if each vertex is above or below the value.  It is faster to
+     * use a for loop than map since it avoids function calls. */
+    for (i = lowhigh.length - 1; i >= 0; i -= 1) {
+      lowhigh[i] = mesh.value[i] <= val ? 0 : 1;
+    }
+    var vpe = mesh.verticesPerElement,
+        square = mesh.shape === 'square',
+        elem = mesh.elements,
+        elemLen = elem.length;
+    for (v = 0; v < elemLen; v += vpe) {
+      if (square) {
+        pattern = lowhigh[elem[v]] * 8 + lowhigh[elem[v + 1]] * 4 +
+                  lowhigh[elem[v + 2]] * 2 + lowhigh[elem[v + 3]];
+        if (pattern === 0 || pattern === 15) {
+          continue;
+        }
+      } else {
+        pattern = 16 + lowhigh[elem[v]] * 4 + lowhigh[elem[v + 1]] * 2 +
+                  lowhigh[elem[v + 2]];
+        if (pattern === 16 || pattern === 23) {
+          continue;
+        }
+      }
+      patternLineTable[pattern].forEach(function (lineEntry) {
+        chains = m_this._addSegment(
+          chains,
+          [elem[v + lineEntry.l0], elem[v + lineEntry.h0]],
+          [elem[v + lineEntry.l1], elem[v + lineEntry.h1]]
+        );
+      });
+    }
+    /* convert chains to lines */
+    lines = chains.map(function (chain) {
+      var line = [];
+      chain.forEach(function (vertex) {
+        var v = m_this._chainVertex(mesh, value, vertex);
+        if (!line.length || v.x !== line[line.length - 1].x ||
+            v.y !== line[line.length - 1].y) {
+          line.push(v);
+        }
+      });
+      line.closed = (line[0].x === line[line.length - 1].x &&
+                     line[0].y === line[line.length - 1].y);
+      /* Add value, level, position, and label information to the line. */
+      line.value = value.value;
+      line.level = value.level;
+      line.position = value.position;
+      line.label = value.label;
+      return line;
+    }).filter(function (line) { return line.length > 1; });
+    return lines;
+  };
+
+  /**
+   * When the feature is marked as modified, mark our sub-feature as
+   * modified, too.
+   *
+   * @returns {object} The results of the superclass modified function.
+   */
+  this.modified = function () {
+    var result = s_modified();
+    if (m_lineFeature) {
+      m_lineFeature.modified();
+    }
+    if (m_labelFeature) {
+      m_labelFeature.modified();
+    }
+    return result;
+  };
+
+  /**
+   * Compute the positions for labels on each line.  This can be called to
+   * recompute label positions without needign to recompute isolines, for
+   * instance when the zoom level changes.  Label positions are computed in the
+   * map gcs coordinates, not interface gcs coordinates, since the interface
+   * gcs may not be linear with the display space.
+   *
+   * @returns {this}
+   */
+  this.labelPositions = function () {
```

Comment:
Do we need this as a public function?

Reply:
Yes.  By default, we recompute local positions after some amount of zooming or panning.  If a user wants the labels to be updated on different criteria, they would call this function.